pic-sure-resources/pic-sure-resource-api/src/main/java/edu/harvard/dbmi/avillach/domain/GICQueryRequest.java

```diff
@@ -0,0 +1,123 @@
+package edu.harvard.dbmi.avillach.domain;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@Schema(name = "QueryRequest", description = "Object containing credentials map under 'resourceCredentials' and query" +
+		" object under 'query'. The query object expectedResultType can be on of the following " +
+		"\"COUNT\", \"CROSS_COUNT\", \"INFO_COLUMN_LISTING\", \"OBSERVATION_COUNT\", \"OBSERVATION_CROSS_COUNT\", \"DATAFRAME\". ",
+		example = "{\n" +
+				"    \"resourceUUID\": \"<RESOURCE UUID>\",\n" +
+				"    \"query\": {\n" +
+				"        \"categoryFilters\": {\n" +
+				"            \"\\\\demographics\\\\SEX\\\\\": [\n" +
+				"                \"female\",\n" +
+				"                \"male\"\n" +
+				"            ]\n" +
+				"        },\n" +
+				"        \"numericFilters\": {\n" +
+				"            \"\\\\demographics\\\\AGE\\\\\": {\n" +
+				"                \"min\": \"0\",\n" +
+				"                \"max\": \"85\"\n" +
+				"            }\n" +
+				"        },\n" +
+				"        \"requiredFields\": [],\n" +
+				"        \"anyRecordOf\": [\n" +
+				"            \"\\\\demographics\\\\RACE\\\\\"\n" +
+				"        ],\n" +
+				"        \"variantInfoFilters\": [\n" +
+				"            {\n" +
+				"                \"categoryVariantInfoFilters\": {},\n" +
+				"                \"numericVariantInfoFilters\": {}\n" +
+				"            }\n" +
+				"        ],\n" +
+				"        \"expectedResultType\": \"DATAFRAME\",\n" +
+				"        \"fields\": []\n" +
+				"    }\n" +
+				"}")
+
+/*
+ * QueryRequests for vanilla GIC PIC-SURE. Includes details about the source
+ * of the request, the requester, and any unifying identifiers for sharing
+ */
+public class GICQueryRequest implements QueryRequest {
```

Comment:
Can we name this something more generic? The implementation of a common area may one day be applied outside of GIC.